src/main/java/seedu/address/ui/cards/InternshipCard.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.ui.cards;
+
+import static seedu.address.ui.GuardClauseUi.IS_EMPTY_LIST_STRING;
+import static seedu.address.ui.panel.PanelDisplayKeyword.COMPANY_DISPLAY_NAME;
+import static seedu.address.ui.panel.PanelDisplayKeyword.JOB_TITLE_DISPLAY_NAME;
+import static seedu.address.ui.panel.PanelDisplayKeyword.PERIOD_DISPLAY_NAME;
+import static seedu.address.ui.panel.PanelDisplayKeyword.REQUIREMENTS_DISPLAY_NAME;
+import static seedu.address.ui.panel.PanelDisplayKeyword.WAGE_DISPLAY_NAME;
+
+import seedu.address.model.internship.InternshipItem;
+
+/**
+ * A UI component that displays some information of a {@code InternshipItem}.
+ */
+public class InternshipCard extends Card<InternshipItem> {
+
+    /**
+     * Creates a card with information regarding {@code InternshipItem} with index of {@code displayedIndex}.
+     *
+     * @param internshipItem The internship item to be displayed.
+     * @param displayedIndex The index of the item.
+     */
+    public InternshipCard(InternshipItem internshipItem, int displayedIndex) {
+        super(internshipItem, displayedIndex);
+        initializedInternshipCardGui();
+    }
+
+    /**
+     * sets the id, company name, requirements, job title, wage, period.
+     */
+    private void initializedInternshipCardGui() {
+        setId(displayedIndex);
+        setCompanyName();
+        setRequirements();
+        setJobTitle();
+        setWage();
+        setPeriod();
+    }
+
+    /**
+     * Sets the company name on the card.
+     */
+    private void setCompanyName() {
+        Object companyName = mapping.get(COMPANY_DISPLAY_NAME);
+        setName(companyName.toString());
+    }
```

Comment:
The name of the card should be the job title instead of the company name?

Reply:
fixed